internal/net/http/json/json_test.go

```diff
@@ -0,0 +1,512 @@
+package json
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+
+	"github.com/vdaas/vald/internal/log"
+	"github.com/vdaas/vald/internal/net/http/rest"
+)
+
+func TestEncode(t *testing.T) {
+	type args struct {
+		w    io.Writer
+		data interface{}
+	}
+
+	type test struct {
+		name      string
+		args      args
+		checkFunc func(err error) error
+	}
+
+	tests := []test{
+		func() test {
+			buf := new(bytes.Buffer)
+			data := map[string]string{
+				"name": "vald",
+			}
+
+			return test{
+				name: "write encoded object",
+				args: args{
+					w:    buf,
+					data: data,
+				},
+				checkFunc: func(err error) error {
+					if err != nil {
+						return fmt.Errorf("err not equals. want: %v, got: %v", nil, err)
+					}
+
+					if got, want := buf.String(), "{\"name\":\"vald\"}\n"; got != want {
+						return fmt.Errorf("output data not equals. want: %v, got: %v", want, got)
+					}
+
+					return nil
+				},
+			}
+		}(),
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := Encode(tt.args.w, tt.args.data)
+			if err := tt.checkFunc(err); err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+func TestDecode(t *testing.T) {
+	type args struct {
+		r    io.Reader
+		data map[string]string
+	}
+
+	type test struct {
+		name      string
+		args      args
+		checkFunc func(err error, data map[string]string) error
+	}
+
+	tests := []test{
+		func() test {
+			buf := new(bytes.Buffer)
+			buf.WriteString(`{"name":"vald"}`)
+
+			return test{
+				name: "read encoded objecg",
+				args: args{
+					r:    buf,
+					data: make(map[string]string, 1),
+				},
+				checkFunc: func(err error, data map[string]string) error {
+					if err != nil {
+						return fmt.Errorf("err not equals. want: %v, got: %v", nil, err)
+					}
+
+					if got, want := data, map[string]string{
+						"name": "vald",
+					}; !reflect.DeepEqual(got, want) {
+						return fmt.Errorf("read data not equals. want: %v, got: %v", want, got)
+					}
+
+					return nil
+				},
+			}
+		}(),
```

Comment:
can you add a case that decode failed?

Reply:
thank you!
But do I need a test that fails decoding ??
Because the implementation code just calls the library, so I think that what kind of data passed will cause an error has been tested in the library:thinking:

```
https://github.com/vdaas/vald/blob/master/internal/net/http/json/json.go#L46
```

Reply:
I fixed it